src/buffer/out/AttrRow.hpp

```diff
@@ -65,7 +72,7 @@ class ATTR_ROW final
     friend class AttrRowIterator;
 
 private:
-    std::vector<TextAttributeRun> _list;
+    boost::container::small_vector<TextAttributeRun, 1> _list;

```

Comment:
Interesting. Should we increase the default loading factor for this guy to 2 or 3? I'm imagining the default Ubuntu shell prompt, which has two colors plus the input area (so, reset to default color). It seems like it would be cheap enough to spare a couple bytes. What do you think?

Reply:
I was just testing the cmd defaults and one was enough there, but if there are shells that need 2 or 3 then maybe this is a good idea.  It's a trade-off of size for every scenario vs. speed for some scenarios.

With my change, each TextAttributeRun is 20 bytes, so given there are 9001 of these created in the default cmd scenario, each increase by 1 in this small_vector costs ~176KB, which seems like enough we should think carefully about what to tune for, but not so much that bumping to 2 or 3 is terrible.

Is there any data on how often different shells are used so we could try to tune for the most common ones?

Reply:
I think it should be 1 because even for Ubuntu, the color is sparse filled in the entire buffer relative to the common case of a single color line. The proportion of 2-3 color prompt lines relative to single color output lines should leave this as the common case still.

Also this code is still going back into Windows and the launch size on the CMD common case is still a big deal there to the perf gates.

Reply:
fair enough!